textile/chat-features.textile

```diff
@@ -9,5 +9,214 @@ jump_to:
+*** @(CHA-RL1h3)@ @[Testable]@ When the room enters the @SUSPENDED@ status as a result of @CHA-RL1h2@ the attach operation shall fail using the @ErrorInfo@ associated with the @SUSPENDED@ room status as the error. The room lifecycle shall asnychronously enter the recovery loop (per "@CHA-RL5@":#CHA-RL5) and attempt to re-establish attachment.
+*** @(CHA-RL1h4)@ @[Testable]@ If the underlying Realtime Channel entered the @FAILED@ state, then the room status will transition to @FAILED@, using the Realtime Channel error as the @cause@ field of the @ErrorInfo@. The same error shall be thrown as the result of the @ATTACH@ operation.
+*** @(CHA-RL1h5)@ @[Testable]@ When the room enters the @FAILED@ status as a result of @CHA-RL1h4@, then the room will detach all channels that are not in the @FAILED@ state.
+*** @(CHA-RL1h6)@ @[Testable]@ When the room enters the @FAILED@ status as a result of @CHA-RL1h4@, and a channel fails to detach on command, then the detach operation will be repeated until such a time that all channels have detached successfully.
+** @(CHA-RL1i)@ Because of the singleton pattern that the core SDKs implement with regards to channels, subsequent instances of a room will use the same underlying realtime channels if the realtime channels are not properly @released@ as part of cleaning up the previous room instance before the new instance is used. Therefore, it is important to ensure that releasing operations (per "@CHA-RC1d@"#CHA-RC1d) have completed before a new room can be attached (to avoid channel objects leaking between instances of rooms).
+*** @(CHA-RL1i1)@ @[Testable]@ If a room instance is in the process of being released and cleaned up (per "@CHA-RC1d@"#CHA-RC1d), then the @ATTACH@ operation of a subsequent instance of the same room shall wait for the release operation of the previous to complete before commencing.
+* @(CHA-RL2)@ A room must be explicitly detached via the @DETACH@ operation.
+** @(CHA-RL2a)@ @[Testable]@ If the room status is already @DETACHED@, then this operation is a no-op.
+** @(CHA-RL2b)@ @[Testable]@ If the room is in the @RELEASING@ status, the operation shall be rejected with an @ErrorInfo@ with code @102102@.
+** @(CHA-RL2c)@ @[Testable]@ If the room is in the @RELEASED@ status, the operation shall be rejected with an @ErrorInfo@ with code @102103@.
+** @(CHA-RL2d)@ @[Testable]@ If the room is in the @FAILED@ status, the operation shall be rejected with an @ErrorInfo@ with code @102101@.
+** @(CHA-RL2e)@ @[Testable]@ Notwithstanding the above points, when the detachment operation begins, the room shall be transitioned to the @DETACHING@ status and any transient disconnect timeouts cleared.
+** @(CHA-RL2f)@ @[Testable]@ The underlying @contributors@ Realtime Channels will be detached in sequence.
```

Comment:
I think that all of the language improvements / clarifications that you made for ATTACH (e.g. what does "in sequence" mean) apply equally to the very-similar language in DETACH.